<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from box_robot.xacro                | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="box_robot">
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="dark">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <material name="light_black">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="grey">
    <color rgba="0.5 0.5 0.5 1.0"/>
  </material>
  <material name="orange">
    <color rgba="1.0 0.4235294117647059 0.0392156862745098 1.0"/>
  </material>
  <material name="brown">
    <color rgba="0.8705882352941177 0.8117647058823529 0.7647058823529411 1.0"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <!-- http://wiki.ros.org/simulator_gazebo/Tutorials/ListOfMaterials -->
  <gazebo reference="base_link">
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="steer_front_left">
    <material>Gazebo/Green</material>
  </gazebo>
  <gazebo reference="steer_front_right">
    <material>Gazebo/Red</material>
  </gazebo>
  <gazebo reference="steer_rear_left">
    <material>Gazebo/Green</material>
  </gazebo>
  <gazebo reference="steer_rear_right">
    <material>Gazebo/Green</material>
  </gazebo>
  <gazebo reference="wheel_front_left">
    <material>Gazebo/Grey </material>
  </gazebo>
  <gazebo reference="wheel_front_right">
    <material>Gazebo/Grey </material>
  </gazebo>
  <gazebo reference="wheel_rear_left">
    <material>Gazebo/Grey </material>
  </gazebo>
  <gazebo reference="wheel_rear_right">
    <material>Gazebo/Grey </material>
  </gazebo>
  <gazebo>
    <plugin filename="libcontrol_plugin.so" name="control_plugin"/>
  </gazebo>
  <!-- <material name="black">
    <color rgba="0.0 0.0 0.0 1.0" />
  </material>

  <material name="dark">
    <color rgba="0.3 0.3 0.3 1.0" />
  </material>

  <material name="light_black">
    <color rgba="0.4 0.4 0.4 1.0" />
  </material>

  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0" />
  </material>

  <material name="green">
    <color rgba="0.0 0.8 0.0 1.0" />
  </material>

  <material name="grey">
    <color rgba="0.5 0.5 0.5 1.0" />
  </material>

  <material name="orange">
    <color rgba="${255/255} ${108/255} ${10/255} 1.0" />
  </material>

  <material name="brown">
    <color rgba="${222/255} ${207/255} ${195/255} 1.0" />
  </material>

  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0" />
  </material>

  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0" />
  </material> -->
  <!-- ================== -->
  <!-- 0) Properties      -->
  <!-- ================== -->
  <!-- ================== -->
  <!-- 1) Base Link       -->
  <!-- ================== -->
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.5 0.5 0.5"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.5 0.5 0.5"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
    </inertial>
  </link>
  <!-- ============== -->
  <!-- 2) Wheel       -->
  <!-- ============== -->
  <link name="steer_front_left">
    <visual>
      <geometry>
        <box size="0.05 0.05 0.1"/>
      </geometry>
      <material name="green"/>
    </visual>
    <!-- <collision>
        <geometry>
          <box size="${steer_size}" />
        </geometry>
      </collision> -->
    <!-- <inertial>
        <mass value="${steer_mass}" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1" />
      </inertial> -->
  </link>
  <link name="wheel_front_left">
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0005" ixy="0" ixz="0" iyy="0.0005" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="steer_front_left_joint" type="revolute">
    <parent link="base_link"/>
    <child link="steer_front_left"/>
    <origin rpy="0 0 0" xyz="-0.2 0.2 -0.3"/>
    <axis xyz="0 0 1"/>
    <limit effort="10.0" lower="-0.785" upper="0.785" velocity="2.0"/>
  </joint>
  <joint name="wheel_front_left_joint" type="continuous">
    <parent link="steer_front_left"/>
    <child link="wheel_front_left"/>
    <origin rpy="0 1.5708 0" xyz="0 0 -0.05"/>
    <axis xyz="0 0 1"/>
    <!-- <limit lower="-0.785" upper="0.785" effort="10.0" velocity="2.0" /> -->
  </joint>
  <link name="steer_front_right">
    <visual>
      <geometry>
        <box size="0.05 0.05 0.1"/>
      </geometry>
      <material name="green"/>
    </visual>
    <!-- <collision>
        <geometry>
          <box size="${steer_size}" />
        </geometry>
      </collision> -->
    <!-- <inertial>
        <mass value="${steer_mass}" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1" />
      </inertial> -->
  </link>
  <link name="wheel_front_right">
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0005" ixy="0" ixz="0" iyy="0.0005" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="steer_front_right_joint" type="revolute">
    <parent link="base_link"/>
    <child link="steer_front_right"/>
    <origin rpy="0 0 0" xyz="0.2 0.2 -0.3"/>
    <axis xyz="0 0 1"/>
    <limit effort="10.0" lower="-0.785" upper="0.785" velocity="2.0"/>
  </joint>
  <joint name="wheel_front_right_joint" type="continuous">
    <parent link="steer_front_right"/>
    <child link="wheel_front_right"/>
    <origin rpy="0 1.5708 0" xyz="0 0 -0.05"/>
    <axis xyz="0 0 1"/>
    <!-- <limit lower="-0.785" upper="0.785" effort="10.0" velocity="2.0" /> -->
  </joint>
  <link name="steer_rear_left">
    <visual>
      <geometry>
        <box size="0.05 0.05 0.1"/>
      </geometry>
      <material name="green"/>
    </visual>
    <!-- <collision>
        <geometry>
          <box size="${steer_size}" />
        </geometry>
      </collision> -->
    <!-- <inertial>
        <mass value="${steer_mass}" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1" />
      </inertial> -->
  </link>
  <link name="wheel_rear_left">
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0005" ixy="0" ixz="0" iyy="0.0005" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="steer_rear_left_joint" type="revolute">
    <parent link="base_link"/>
    <child link="steer_rear_left"/>
    <origin rpy="0 0 0" xyz="-0.2 -0.2 -0.3"/>
    <axis xyz="0 0 1"/>
    <limit effort="10.0" lower="-0.785" upper="0.785" velocity="2.0"/>
  </joint>
  <joint name="wheel_rear_left_joint" type="continuous">
    <parent link="steer_rear_left"/>
    <child link="wheel_rear_left"/>
    <origin rpy="0 1.5708 0" xyz="0 0 -0.05"/>
    <axis xyz="0 0 1"/>
    <!-- <limit lower="-0.785" upper="0.785" effort="10.0" velocity="2.0" /> -->
  </joint>
  <link name="steer_rear_right">
    <visual>
      <geometry>
        <box size="0.05 0.05 0.1"/>
      </geometry>
      <material name="green"/>
    </visual>
    <!-- <collision>
        <geometry>
          <box size="${steer_size}" />
        </geometry>
      </collision> -->
    <!-- <inertial>
        <mass value="${steer_mass}" />
        <origin xyz="0 0 0" />
        <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1" />
      </inertial> -->
  </link>
  <link name="wheel_rear_right">
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.04" radius="0.07"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0005" ixy="0" ixz="0" iyy="0.0005" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="steer_rear_right_joint" type="revolute">
    <parent link="base_link"/>
    <child link="steer_rear_right"/>
    <origin rpy="0 0 0" xyz="0.2 -0.2 -0.3"/>
    <axis xyz="0 0 1"/>
    <limit effort="10.0" lower="-0.785" upper="0.785" velocity="2.0"/>
  </joint>
  <joint name="wheel_rear_right_joint" type="continuous">
    <parent link="steer_rear_right"/>
    <child link="wheel_rear_right"/>
    <origin rpy="0 1.5708 0" xyz="0 0 -0.05"/>
    <axis xyz="0 0 1"/>
    <!-- <limit lower="-0.785" upper="0.785" effort="10.0" velocity="2.0" /> -->
  </joint>
</robot>

